pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+        files or remove them with :meth:`~pywbem_mock.DMTFCIMSchema.clean` when
+        not being used.
+
+        If the schema is to be committed a source repository such as git
+        it is logical to commit only the DMTF CIM schema zip file. Creation of
+        the `schema_mof_dir` subdirectory will be created when the
+        :class:`pywbem_mock.DMTFCIMSchema` object is created.
+
+        Raises:
+            ValueError: If the schema cannot be retrieved from the DMTF web
+              site.
+            TypeError:  If the `schema_version` is not a valid tuple with 3
+              integer components
+        """
+
+        def print_verbose(msg):
+            """
+            Inner method prints msg if var `first` is `False` and verbose is
+            `True`. Then it sets first to `False`.
+            This allows displaying steps that are actually executed.
+            """
+            if self.verbose:
+                print(msg)
+
+        if not os.path.isdir(self.schema_root_dir):
+            print_verbose("Creating directory for CIM Schema archive: %s" %
+                          self.schema_root_dir)
+            os.mkdir(self.schema_root_dir)
+
+        if not os.path.isfile(self._mof_zip_filename):
+            print_verbose("Downloading CIM Schema archive from: %s" %
+                          self._mof_zip_url)
+
+            try:
+                ufo = urlopen(self._mof_zip_url)
+            except IOError as ie:
+                os.rmdir(self.schema_root_dir)
+                raise ValueError('DMTF Schema archive not found at url %s: %s' %
+                                 (self._mof_zip_url, ie))
+
+            with open(self._mof_zip_filename, 'wb') as fp:
+                for data in ufo:
+                    fp.write(data)
+
+        if not os.path.isdir(self.schema_mof_dir):
+            print_verbose("Creating directory for CIM Schema MOF files: %s" %
+                          self.schema_mof_dir)
+            os.mkdir(self.schema_mof_dir)
+        if not os.path.isfile(self._schema_mof_filename):
+            print_verbose("Unpacking CIM Schema archive: %s" %
+                          self._mof_zip_filename)
+
+            zfp = None
+            try:
+                zfp = ZipFile(self._mof_zip_filename, 'r')
+                nlist = zfp.namelist()
+                for file_ in nlist:
+                    dfile = os.path.join(self.schema_mof_dir, file_)
+                    if dfile[-1] == '/':
+                        if not os.path.exists(dfile):
+                            os.mkdir(dfile)
+                    else:
+                        with open(dfile, 'w+b') as dfp:
+                            dfp.write(zfp.read(file_))
+            finally:
+                if zfp:
+                    zfp.close()
+
+    def build_schema_mof(self, schema_classes):
+        """
+        Build a string that includes the ``#include pragmas`` for the DMTF
```

Comment:
DOn't put plural on the `pragma` keyword, but say "`#include pragma` statements"

Reply:
DONE